src/include/logging/phylog_checkpoint_manager.h

```diff
@@ -0,0 +1,62 @@
+//===----------------------------------------------------------------------===//
+//
+//                         Peloton
+//
+// phylog_checkpoint_manager.h
+//
+// Identification: src/backend/logging/checkpoint/phylog_checkpoint_manager.h
+//
+// Copyright (c) 2015-16, Carnegie Mellon University Database Group
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include "logging/checkpoint_manager.h"
+
+namespace peloton {
+namespace logging {
+
+/**
+ * checkpoint file name layout :
+ * 
+ * dir_name + "/" + prefix + "_" + checkpointer_id + "_" + database_id + "_" + table_id + "_" + epoch_id
+ *
+ *
+ * checkpoint file layout :
+ *
+ *  -----------------------------------------------------------------------------
+ *  | tuple_1 | tuple_2 | tuple_3 | ...
+ *  -----------------------------------------------------------------------------
+ *
+ * NOTE: tuple length can be obtained from the table schema.
+ *
+ */
+
+class PhyLogCheckpointManager : public CheckpointManager {
+  // Deleted functions
+  PhyLogCheckpointManager(const PhyLogCheckpointManager &) = delete;
+  PhyLogCheckpointManager &operator=(const PhyLogCheckpointManager &) = delete;
+  PhyLogCheckpointManager(PhyLogCheckpointManager &&) = delete;
+  PhyLogCheckpointManager &operator=(const PhyLogCheckpointManager &&) = delete;
+
+
+public:
+  PhyLogCheckpointManager() {}
+  virtual ~PhyLogCheckpointManager() {}
+
+  static PhyLogCheckpointManager& GetInstance() {
```

Comment:
We should probably start disallowing these singleton methods. It is starting to get out of control.